### PR TITLE
Feat/#60 modal interaction

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

- Edit User Modal static UI and data hookup
- Broke program manage page Users and Profile tabs out into their own components

Please note:
- success toast still needs to be added, preferably after #149 
- gateway queries still need to be verified

**Type of Change**
- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches deisgn:
* component sizes, spacing, and styless
* font size, weight, colour
* spelling has been double checked
- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [ ] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: client/components/pages/submission-system/program-management/Users.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { ModalPortal } from '../layout';
+import { useSubmitFormHook, createUserInput } from './';
+import EDIT_USER_MUTATION from './EDIT_USER_MUTATION.gql';
+
+const Users = ({ users, shortName }) => {
+  const initialState = { selectedUser: null, showModal: false };
+  const reducer = (state, action) =>
+    action.showModal
+      ? { selectedUser: action.user, showModal: true }
+      : { selectedUser: null, showModal: false };
```
Comment (hlminh2000) [inline client/components/pages/submission-system/program-management/Users.js]: If we use reducers, I recommend designing actions in the shape `{type: string, payload: any}`, then switching on the `type` field to return new state.

Though reducers are probably more useful for dealing with a very large state object that needs very strict control over the available mutations to it, and/or when there's a need for capturing mutations in serializable format, so it's a little overkill here and is adding quite a bit of noise. 

It actually looks like `!!selectedUser` seems like the only check we need to decide whether to show the modal, so the `showModal` state isn't needed either. I think we can name `selectedUser` something more like: `currentEdittingUser`, then just check on `!!currentEdittingUser` for the modal. Then later we can have a `currentDeleteingUser` then we can do the same thing to show a different modal. Can be easily done with two `useState` :)

What do you think?

File: client/components/pages/submission-system/program-management/index.js [hunk 2]
```diff
@@ -121,7 +113,7 @@ export default ({ logOut, pathname }: { logOut: any => any, pathname: string })
               )}
             </Tab>
           </Tabs>
-          {activeTab === TABS.USERS && <Users users={FAKE_USERS} />}
+          {activeTab === TABS.USERS && <Users shortName={shortName} users={FAKE_USERS} />}
```
Comment (hlminh2000) [inline client/components/pages/submission-system/program-management/index.js]: maybe `shortName` prop should be `programShortName`. The current name makes sense for us now but it kinda reads like `shortName` is some kind of user identifier, which can get a bit confusing in future. Inside the `Users` component, the name `shortName` also loses the context that it's a program identifier.

File: client/components/pages/submission-system/program-management/index.js [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
 };
 
+export const createUserInput = ({ data, shortName }) => ({
+  programShortName: shortName,
+  userFirstName: data.firstName,
+  userLastName: data.lastName,
+  userEmail: data.email,
+  userRole: data.role,
+});
+
```
Comment (ciaranschutte) [inline client/components/pages/submission-system/program-management/index.js]: will move these to common file in standardise form ticket: #251

File: client/components/pages/submission-system/program-management/index.js [hunk 4]
```diff
@@ -1,34 +1,34 @@
...
+import Profile from './Profile';
 
-const useSubmitFormHook = ({ gql }) => {
+export const useSubmitFormHook = ({ gql }) => {
   const [triggerMutation, rest] = useMutation(gql);
 
   return [triggerMutation];
 };
 
+export const createUserInput = ({ data, shortName }) => ({
```
Comment (hlminh2000) [inline client/components/pages/submission-system/program-management/index.js]: let's call `shortName` `programShortName`

File: client/components/pages/submission-system/program-management/Users.js (hunk 1 above)
Comment (ciaranschutte) [reply client/components/pages/submission-system/program-management/Users.js]: Yup sounds good.

Removing the redundant state of showModal actually removes the need for the reducer anyway.
I was using it to change both state at the same time.

**Comment by ciaranschutte** [comment]
@hlminh2000 addressed your feedback
Merged in Xu's resent modal which has a lot of state in it like I had before eg.
```
  const handleResendEmailClick = ({ user }) => {
    setUser(user);
    setIsResendEmailModalOpen(true);
  };
```

I can address this in another PR?

**Comment by ciaranschutte** [comment]
@hlminh2000 sounds good (that's why I left it out of my Edit Modal code for now) :)

**Pull request merged**